### PR TITLE
Add PhpMnd task

### PR DESCRIPTION
**Pull request by jyggen**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Documented?   | yes
| Fixed tickets | n/a

This PR adds support for [PHP Magic Number Detector](https://github.com/povils/phpmnd) (PHPMND).

> `phpmnd` is a tool that **helps** you detect magic numbers in PHP code.
>
> **What is Magic Number?**
>Magic number is a numeric literal that is not defined as a constant that may change at a later stage, but that can be therefore hard to update. It is a bad programming practice of using numbers directly in source code without explanation. In most cases this makes programs harder to read, understand, and maintain.

# New Task Checklist:

- [x] Is the README.md file updated?
- [x] Are the dependencies added to the composer.json suggestions?
- [x] Is the doc/tasks.md file updated?
- [x] Are the task parameters documented?
- [x] Is the task registered in the tasks.yml file?
- [x] Does the task contains phpspec tests?
- [x] Is the configuration having logical allowed types?
- [x] Does the task run in the correct context?
- [x] Is the `run()` method readable?
- [x] Is the `run()` method using the configuration correctly?
- [x] Are all CI services returning green?

### CODE REVIEW

File: src/Task/PhpMnd.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    /**
+     * @return OptionsResolver
+     */
+    public function getConfigurableOptions()
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'directory' => '.',
```
Comment (veewee) [inline src/Task/PhpMnd.php]: Won't this also run on ignored folders like vendor by default? Maybe that is something that can be added to exclude_path by default if there is no way to pass files to the command?

**Comment by veewee** [comment]
Thanks @jyggen! The PR looks good, I've added one little question.

File: src/Task/PhpMnd.php (hunk 1 above)
Comment (jyggen) [inline src/Task/PhpMnd.php]: Nice catch! I looked into it and it turns out that PHPMND automatically adds `vendor` to the exclude list ([as seen here](https://github.com/povils/phpmnd/blob/a85c7c746c52f15b8326596523889b16b39a7665/src/PHPFinder.php#L32)), so it should be fine as-is.

**Comment by veewee** [comment]
Great. Thanks for the addition!

**Pull request merged**